vscode-dotnet-runtime-library/src/Acquisition/DotnetCoreAcquisitionWorker.ts

```diff
@@ -79,8 +82,14 @@ export class DotnetCoreAcquisitionWorker implements IDotnetCoreAcquisitionWorker
 
         this.removeFolderRecursively(this.context.installDirectoryProvider.getStoragePath());
 
-        await this.context.extensionState.update(this.installingVersionsKey, []);
-        await this.context.extensionState.update(this.installedVersionsKey, []);
+        // This does not uninstall global things yet, so don't remove their keys.
+        const installingVersions = this.context.extensionState.get<string[]>(this.installingVersionsKey, []);
+        const remainingInstallingVersions = installingVersions.filter(x => this.isGlobalInstallKey(x));
+        await this.context.extensionState.update(this.installingVersionsKey, remainingInstallingVersions);
+
+        const installedVersions = this.context.extensionState.get<string[]>(this.installedVersionsKey, []);
+        const remainingInstalledVersions = installedVersions.filter(x => this.isGlobalInstallKey(x));
```

Comment:
Does here repetitively define the `installedVersions` and `remainingInstallingVerisons` twice?  and call `this.context.extensionState.update` a second time?

Reply:
That's a good question. They are a bit different. There are keys for 'installed' versions and 'installing' versions and those are separate, since one maintains things that have not yet finished installing, and the other is for successfully completed installs in the past. The variable names are rather similar which could probably be improved.